vertx-web/src/main/java/io/vertx/ext/web/handler/sse/EventSource.java

```diff
@@ -0,0 +1,39 @@
+package io.vertx.ext.web.handler.sse;
+
+import io.vertx.codegen.annotations.Fluent;
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Handler;
+import io.vertx.core.Vertx;
+import io.vertx.core.http.HttpClientOptions;
+import io.vertx.ext.web.handler.sse.impl.EventSourceImpl;
+
+public interface EventSource {
+
+	static EventSource create(Vertx vertx, HttpClientOptions options) {
+		return new EventSourceImpl(vertx, options);
+	}
+
+	@Fluent
+  EventSource connect(String path, Handler<AsyncResult<Void>> handler);
+
+	@Fluent
+	default EventSource close() {
+		return null;
+	}
+
+	@Fluent
+  EventSource connect(String path, String lastEventId, Handler<AsyncResult<Void>> handler);
+
+	@Fluent
+  EventSource onMessage(Handler<String> messageHandler);
+
+	@Fluent
+  EventSource onEvent(String eventName, Handler<String> handler);
+
+	@Fluent
+	default EventSource onClose(Handler<Void> handler) {
```

Comment:
Shouldn't this be invoked only when the server returns `204 No Content`?

Reply:
What if the server closes the connection?

Reply:
What does the browser `EventSource` do when the server closes connection?

Reply:
I think there's no such thing as `onclose` on the browser's `EventSource` (or I can't find it), should we get rid of it? Not sure why I added it in the first place (this was a long time ago).

Reply:
Let's stick to the browser `EventSource` behavior and we can refine later if/when needed.